functions/source/ConfigParser/params.py

```diff
@@ -0,0 +1,106 @@
+import urllib.parse
+from typing import Dict
+from dataclasses import dataclass
+import json
+import re
+import boto3
+
+########################
+##  Helper Functions  ##
+########################
+
+# Input is a string that may or may not be URL encoded
+# This function returns the string, as-is, if not URL encoded
+# Returns decoded if is URL encoded
+def safe_decode(string):
+    try_decode = urllib.parse.unquote(string)
+    if len(string) > len(try_decode):
+        return try_decode
+    return string
+
+def is_http(string):
+    pattern = r"^http(s):\/\/.*"
+    is_match = bool(re.match(pattern,string))
+    return is_match
+
+def is_s3(string):
+    pattern = r"^s3:\/\/.*"
+    is_match = bool(re.match(pattern,string))
+    return is_match
+
+def get_from_http(url):
+    f = requests.get(url)
+    return f.text
+
+def get_from_s3(uri):
+    bucket = uri.split("//")[1].split('/')[0]
+    key_list = uri.split("//")[1].split('/')[1:]
+    file_key = "/".join(key_list)
+    s3_client = boto3.client('s3')
+    obj = s3_client.get_object(Bucket=bucket, Key=file_key)
+    content = json.loads(obj['Body'].read().decode('utf-8'))
+    return content
+
+def get_params_list_from_source(param_string):
+    param_string = safe_decode(param_string)
+    if is_http(param_string):
+        content = get_from_http(param_string)
+        return content
+
+    if is_s3(param_string):
+        content = get_from_s3(param_string)
+        return content
+
+    content = json.loads(param_string)
+    return content
+
+# The Params class contains helper scripts for converting Configurator outputs
+# and Cloudformation inputs between different formats
+
+@dataclass
+class Params:
+    params: Dict
+
+    @staticmethod
+    def get_dict_from_params_input_agnostic(params_input):
+        params = Params.new_from_params_input_agnostic(params_input)
+        params_dict = params.get_dict()
+        return params_dict
+
+    @staticmethod
+    def new_from_params_input_agnostic(params_input):
+        params_list = get_params_list_from_source(params_input)
+        params = Params.load_from_list(params_list)
+        return params
+
+    # Used in the above static method
+    @staticmethod
+    def load_from_list(params_list):
+        params_dict = {}
```

Comment:
if you want to be more concise ¯\_(ツ)_/¯
```suggestion
        params_dict = {k: v for k, v in (item.split("=") for item in params_list)}
```

Reply:
Whoa! Pretty cool! Didn't even think of generating this with list/dictionary comprehension. I'll take the pithy comprehension form 9 out of 10 times. :)